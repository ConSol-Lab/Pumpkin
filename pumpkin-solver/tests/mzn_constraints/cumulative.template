predicate gecode_cumulatives(array [int] of var int: start_times,
    array [int] of var int: durations,
    array [int] of var int: resource_requirements,
    var int: resource_capacity
);

var 1..5: x:: output_var;
var 3..5: y:: output_var;
var 1..10: z:: output_var;

constraint gecode_cumulatives([x, y, z], [1, 2, 3], [1, 2, 1], 2);

solve satisfy;
