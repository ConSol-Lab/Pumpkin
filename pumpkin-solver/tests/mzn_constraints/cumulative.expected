x = 1;
y = 3;
z = 5;
----------
x = 1;
y = 3;
z = 6;
----------
x = 1;
y = 3;
z = 7;
----------
x = 1;
y = 3;
z = 8;
----------
x = 1;
y = 3;
z = 9;
----------
x = 1;
y = 3;
z = 10;
----------
x = 2;
y = 3;
z = 5;
----------
x = 2;
y = 3;
z = 6;
----------
x = 2;
y = 3;
z = 7;
----------
x = 2;
y = 3;
z = 8;
----------
x = 2;
y = 3;
z = 9;
----------
x = 2;
y = 3;
z = 10;
----------
x = 5;
y = 3;
z = 5;
----------
x = 5;
y = 3;
z = 6;
----------
x = 5;
y = 3;
z = 7;
----------
x = 5;
y = 3;
z = 8;
----------
x = 5;
y = 3;
z = 9;
----------
x = 5;
y = 3;
z = 10;
----------
x = 1;
y = 4;
z = 1;
----------
x = 1;
y = 4;
z = 6;
----------
x = 1;
y = 4;
z = 7;
----------
x = 1;
y = 4;
z = 8;
----------
x = 1;
y = 4;
z = 9;
----------
x = 1;
y = 4;
z = 10;
----------
x = 2;
y = 4;
z = 1;
----------
x = 2;
y = 4;
z = 6;
----------
x = 2;
y = 4;
z = 7;
----------
x = 2;
y = 4;
z = 8;
----------
x = 2;
y = 4;
z = 9;
----------
x = 2;
y = 4;
z = 10;
----------
x = 3;
y = 4;
z = 1;
----------
x = 3;
y = 4;
z = 6;
----------
x = 3;
y = 4;
z = 7;
----------
x = 3;
y = 4;
z = 8;
----------
x = 3;
y = 4;
z = 9;
----------
x = 3;
y = 4;
z = 10;
----------
x = 1;
y = 5;
z = 1;
----------
x = 1;
y = 5;
z = 2;
----------
x = 1;
y = 5;
z = 7;
----------
x = 1;
y = 5;
z = 8;
----------
x = 1;
y = 5;
z = 9;
----------
x = 1;
y = 5;
z = 10;
----------
x = 2;
y = 5;
z = 1;
----------
x = 2;
y = 5;
z = 2;
----------
x = 2;
y = 5;
z = 7;
----------
x = 2;
y = 5;
z = 8;
----------
x = 2;
y = 5;
z = 9;
----------
x = 2;
y = 5;
z = 10;
----------
x = 3;
y = 5;
z = 1;
----------
x = 3;
y = 5;
z = 2;
----------
x = 3;
y = 5;
z = 7;
----------
x = 3;
y = 5;
z = 8;
----------
x = 3;
y = 5;
z = 9;
----------
x = 3;
y = 5;
z = 10;
----------
x = 4;
y = 5;
z = 1;
----------
x = 4;
y = 5;
z = 2;
----------
x = 4;
y = 5;
z = 7;
----------
x = 4;
y = 5;
z = 8;
----------
x = 4;
y = 5;
z = 9;
----------
x = 4;
y = 5;
z = 10;
----------
==========
